meilisearch-release-changelog/v1.9.0.md

```diff
@@ -0,0 +1,215 @@
+
+## New feature: Ranking score threshold
+
+Use `rankingScoreThreshold` to exclude search results with low ranking scores:
+
+```bash
+curl \
+ -X POST 'http://localhost:7700/indexes/movies/search' \
+ -H 'Content-Type: application/json' \
+ --data-binary '{
+    "q": "Badman dark returns 1",
+    "showRankingScore": true,
+    "limit": 5,
+    "rankingScoreThreshold": 0.2
+ }'
+```
+
+Meilisearch does not return any documents below the configured threshold. Excluded results do not count towards `estimatedTotalHits`, `totalHits`, and facet distribution.
+
+⚠️ For performance reasons, if the number of documents above `rankingScoreThreshold` is higher than `limit`, Meilisearch does not evaluate the ranking score of the remaining documents. Results ranking below the threshold are not immediately removed from the set of candidates. In this case, Meilisearch may overestimate the count of `estimatedTotalHits`, `totalHits` and facet distribution.
+
+Done by @dureuill in #4666
+
+## New feature: Get similar documents endpoint
+
+This release introduces a new AI-powered search feature allowing you to send a document to Meilisearch and receive a list of similar documents in return.
+
+Use the `/indexes/{indexUid}/similar` endpoint to query Meilisearch for related documents:
+
+```sh
+curl \
+  -X POST /indexes/:indexUid/similar
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "id": "23",
+    "offset": 0,
+    "limit": 2,
+    "filter": "release_date > 1521763199",
+    "embedder": "default",
+    "attributesToRetrieve": [],
+    "showRankingScore": false,
+    "showRankingScoreDetails": false
+  }'
+```
+
+- `id`: string indicating the document needing similar results, required
+- `offset`: number of results to skip when paginating, optional, defaults to `0`
+- `limit`: number of results to display, optional, defaults to `20`
+- `filter`: string with a filter expression Meilisearch should apply to the results, optional, defaults to `null`
+- `embedder`: string indicating the embedder Meilisearch should use to retrieve similar documents, optional, defaults to `"default"`
+- `attributesToRetrieve`: array of strings indicating which fields Meilisearch will include in the response, optional, defaults to `["*"]`
+-  `showRankingScore`: boolean indicating if results should include ranking score information, optional, defaults to `false`
+- `showRankingScoreDetails`: boolean indicating if results should include detailed ranking score information, optional, defaults to `false`
+
+`/indexes/{indexUid}/similar` supports `GET` and `POST` routes. Use URL query parameters to configure your `GET` request, or include your parameters in the request body if using the `POST` route.  Both offer identical functionality.
+
+Done by @dureuill in #4647
+
+## New feature: `frequency` matching strategy
+
+This release adds a new matching strategy, `frequency`. Use it to prioritize results containing the least frequent query terms:
```

Comment:
No, we prioritize the least frequent words because they are the most discriminating ones.
For instance, the query `Soup of the day`, `of` and `the` are really frequent in documents so they are not really useful to rank documents. We would prefer to focus on `Soup` and `Day`.

Reply:
Thanks for the clarification @ManyTheFish. 

May I suggest reconsidering the naming of the option so it aligns better with the naming of others? Something like `discriminate` or `infrequent` would seem more logical. 

@guimachiavelli or @CaroFG may have better suggestions than me 😅

Reply:
IMO `frequency` does not tell if we're using the most frequent or least frequent. From here, the name sounds apt, and I believe it is a term of art too 🤔

Reply:
Wanted to post a longer response, because this is incorrect:

> This naming seems inconsistent with the naming of other options:
> • `all` prioritizes results with all query terms
> • `last` prioritizes results where the last query term is present

Matching strategies don't prioritize results. They define the order in which words are iteratively removed from the query:

- `all`: will remove all words from the query at once, meaning only documents with all words will match.
- `last`: each iteration will remove the last word from the query, meaning that the first iteration requires all words, then the second iteration requires all words minus the last one, etc.
- `frequency`: each iteration will remove the word with the highest frequency from the query, meaning that the first iteration requires all words, then the second iteration requires all words minus the most frequent one, etc.

The name is consistent with this model, which is how the engine handles matching strategies internally.
I think straying from this mental model to go into "prioritizes result" mental models results in a blurry mental model where it is unclear how the matching strategy actually affect the query, and its link with the `words` score details (which displays the number of `maxMatchingWords` and number of remaining `matchingWords`).

In conclusion, I strongly believe the name should remain `frequency`.